extensions/roam-research/src/utils.ts

```diff
@@ -1,29 +1,49 @@
 import { useRef, useCallback } from "react";
 import dayjs from "dayjs";
-import { ReversePullBlock } from "./type";
+import { usePersistentState } from "raycast-toolkit";
 
-export const CONSTANTS = {
-  keys: {
-    graph: "graph-key",
-    "graph-name": (name: string) => "graph-key-" + name,
-  },
+// hook that you should use to get graph settings like name, token
+// this stores it in Raycast's encrypted localstorage
+// note that we DO NOT store this in Raycast's cache because it's unsecure and because it's an LRU cache, it can remove least recently used data when more than 10 MB in cache
+// might want to move this to someplace like `./roamApi.ts`
+export const useGraphsConfig: () => {
+  graphsConfig: GraphsConfigMap;
+  saveGraphConfig: (obj: GraphConfig) => void;
+  removeGraphConfig: (graphName: string) => void;
+  isGraphsConfigLoading: boolean;
+} = () => {
+  // TODO: usePersistentState has an error when even when we do `setGraphsConfig`, `graphsConfig` does not change/cause rerenders
+  const [graphsConfig, setGraphsConfig, isGraphsConfigLoading] = usePersistentState<GraphsConfigMap>(
+    "graphs-config",
+    {}
+  );
+  const saveGraphConfig = (obj: GraphConfig) => {
+    setGraphsConfig({
+      ...graphsConfig,
+      [obj.nameField]: obj,
+    });
+  };
+  const removeGraphConfig = (graphName: string) => {
+    const newGraphsConfig = { ...graphsConfig };
+    delete newGraphsConfig[graphName];
+    setGraphsConfig(newGraphsConfig);
+  };
+  // console.log("Loaded graphsConfig: ", graphsConfig);
+  return { graphsConfig, saveGraphConfig, removeGraphConfig, isGraphsConfigLoading } as const;
 };
 
 export const keys = <T extends object>(obj: T) => {
   return Object.keys(obj) as (keyof T)[];
 };
 
-export const debounce = <T, D>(cb: (...args: T[]) => D) => {
```

Comment:
removed this debounce function because was running into issues with it. Using the npm module "debounce" instead